src/main/java/umaikaze/duke/Storage.java

```diff
@@ -0,0 +1,83 @@
+package umaikaze.duke;
+
+import umaikaze.duke.task.Deadline;
+import umaikaze.duke.task.Event;
+import umaikaze.duke.task.Task;
+import umaikaze.duke.task.Todo;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Storage {
+    String savePath;
+
+    public Storage(String savePath) {
+        this.savePath = savePath;
+    }
+
+    public void saveFile(List<Task> list) throws IOException, DukeException {
+        PrintWriter out = new PrintWriter(savePath, "UTF-8");
+        StringBuilder sb = new StringBuilder("");
+        for (int i = 1; i <= list.size(); i++) {
+            Task task = list.get(i - 1);
+            sb.append(task.getSaveString()).append("\n");
+        }
+        out.print(sb.toString());
+        out.close();
+    }
+
+    public List<Task> loadFile() throws IOException {
+        List<Task> list = new ArrayList<>(100);
+        BufferedReader br = new BufferedReader(new FileReader(savePath));
+        String nextLine = br.readLine();
+        String[] line;
+        if (nextLine != null) {
+            line = nextLine.split("\\|");
+            for (int i = 0; i < line.length; i++) {
+                line[i] = line[i].trim();
+            }
+            while (nextLine != null) {
+                Task task;
+                Parser p;
+                switch (line[0]) {
+                    case "T":
+                        task = new Todo(line[2]);
+                        if (line[1].equals("1")) {
+                            task.markDone();
+                        }
+                        System.out.println("\tLoaded a todo");
```

Comment:
Consider using an appropriate method in your `Ui` class to do the printing, instead of directly doing `System.out.println`as doing so breaks abstraction.